### PR TITLE
Porting to HPX 1.6.0

### PR DESCRIPTION
**Pull request by albestro**
This PR aims at collecting all required changes for porting to HPX 1.6.0 which, AFAIK, will be released early next year.

I don't think there is any doubt about updating, since we are already looking for it for new features and improvements.

### CODE REVIEW

**Comment by albestro** [comment]
bors try

just checking that CI works using HPX@master

**Comment by albestro** [comment]
bors -

**Comment by albestro** [comment]
bors try-

**Comment by albestro** [comment]
bors try

just checking that CI works using HPX@master

(I missed updating some usages)

**Comment by albestro** [comment]
bors try

try again, just to see if CI works correctly

**Comment by albestro** [comment]
bors try

let's see if it works now (I was putting HPX version in the wrong place)

**Comment by albestro** [comment]
I manually re-run just one of the test and it worked.

Currently there is no need to run a full test, it is enough knowing that the CI (should) run correctly (`codecov cpu` finished all green)

**Comment by msimberg** [comment]
Thank you @albestro for doing this. I'd say this flag should no longer be set if we require 1.6.0: https://github.com/eth-cscs/DLA-Future/blob/4188ddf42e5661961c4882ab712034971d62f882/ci/docker/release/deploy.Dockerfile#L31. I think you've already taken care of the main deprecation which is the priority enum change.

We will definitely be turning off thread executors compatibility by default (which means no executors in `hpx::threads::executors`) and probably the `pool_executor` compatibility, both of which mean that `pool_executor("default")` should be replaced with `hpx::execution::parallel_executor(hpx::resource::get_pool("default"))` (`parallel_executor` also takes priorities and hints nowadays). The MPI pool check could also now use `hpx::resource::pool_exists`.

@albestro please let me know if you'd like me to apply any of these changes.

**Comment by albestro** [comment]
> @albestro please let me know if you'd like me to apply any of these changes.

Who better than you knows about deprecation and changes?!?! 😁 feel free to contribute directly to this PR!

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
Updated the HPX version to 1.6.0-rc1.

bors try

**Comment by albestro** [comment]
not related, but it shows up also here. is it worth extending a bit the allocation time for CI?

@msimberg @rasolca

**Comment by msimberg** [comment]
> not related, but it shows up also here. is it worth extending a bit the allocation time for CI?
> 
> @msimberg @rasolca

No particular opinion on this but if the tests otherwise time out, then yes! Do we know if it's more a temporary issue on the system, or that the tests are starting to take longer in general?

**Comment by msimberg** [comment]
bors try

**Comment by rasolca** [comment]
> > not related, but it shows up also here. is it worth extending a bit the allocation time for CI?
> > @msimberg @rasolca
> 
> No particular opinion on this but if the tests otherwise time out, then yes! Do we know if it's more a temporary issue on the system, or that the tests are starting to take longer in general?

Tests sometimes timeout due to temporary issues on the system.
Anyway currently CI is not working (due to a change in gitlab container registry, sarus cannot pull the images).

**Comment by rasolca** [comment]
bors try

File: spack/packages/dla-future/package.py [hunk 1]
```diff
@@ -26,8 +26,7 @@ class DlaFuture(CMakePackage, CudaPackage):
     depends_on("blaspp")
     depends_on("lapackpp")
     depends_on("hpx cxxstd=14 networking=none")
-    depends_on("hpx@1.5.0", when="~cuda")
-    depends_on("hpx@master +cuda", when="+cuda")
+    depends_on("hpx@1.6.0:")
```
Comment (albestro) [inline spack/packages/dla-future/package.py:29]: @msimberg does it sound correct to you?

Comment (msimberg) [reply spack/packages/dla-future/package.py:29]: It does! Thanks for updating that!

**Comment by msimberg** [comment]
FYI: https://github.com/spack/spack/pull/21757. Once that's in we can go ahead with this.

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
bors try

**Review by teonnik — APPROVED** [review]

**Review by rasolca — APPROVED** [review]

**Pull request merged**